test/vcr_python_wrapper.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import vcr
+import json
+
+VCR_PARAMS_FILE = '../../.vcr_params.json'
+
+sys.argv.pop(0)
+
+# Load recording parameters from file
+with open(VCR_PARAMS_FILE, 'r') as params_file:
+    test_params = json.load(params_file)
+cassette_file = 'fixtures/{}-{}.yml'.format(test_params['test_name'], test_params['serial'])
+# Increase serial and dump back to file
+test_params['serial'] += 1
```

Comment:
We'll definitely need to change this :laughing: The vcr file should describe the current state of the server. Something like 'organization_absent.yml' or 'product_present.yml'. Though, I know you did this just to get it working.

Reply:
If you find a method to get ansible to pass the name of the task to the module, this would be easy. Right now, it depends on the playbook being run in a specific order.

Reply:
Would you be able to derive it from ```sys.argv[0]```?

Reply:
that would be something like `ansible_module_foreman_ptable.py`. So sadly no. And deriving it from the module name and state, would not be enough information.
I would prefer something like:
product_present_created.yml
product_present_updated.yml
product_present_unchanged.yml
product_absent_deleted.yml
and so on...
By the way: Can you make a Playbook task fail on changed==True?

Reply:
Yes, if you use sys.argv[0] (after pop(0)), you'll get fixtures named like this `ansible_module_foreman_organization.py.yml`. Then if we could index those, we'd have `ansible_module_foreman_organization.pyX.yml` where X is in [0, 1, 2, ...].

Reply:
Then we'd have the name of the module, but we need the file for counting and record_mode anyway (not much gained IMHO).
It'd really like to call the fixtures the same as what is given as the `name` of the individual task.